controllers/update_status.go

```diff
@@ -624,9 +640,77 @@ func validateProcessGroup(ctx context.Context, r *FoundationDBClusterReconciler,
 	processGroupStatus.UpdateCondition(fdbv1beta2.PodFailing, failing, cluster.Status.ProcessGroups, processGroupStatus.ProcessGroupID)
 	processGroupStatus.UpdateCondition(fdbv1beta2.PodPending, false, cluster.Status.ProcessGroups, processGroupStatus.ProcessGroupID)
 
+	disableTaintFeature := false
+	for _, taintConfiguredKey := range cluster.Spec.AutomationOptions.Replacements.TaintReplacementOptions {
+		if *taintConfiguredKey.Key == "*" && *taintConfiguredKey.DurationInSeconds < 0 {
```

Comment:
We have the `pointer` library and we should make use of it, otherwise this could led to a nil pointer exception. The pointer library has some methods like `pointer.StringDeref(taintConfiguredKey.Key , "")` to prevent nil pointer errors.